transport/http_test.go

```diff
@@ -254,3 +208,105 @@ func TestHTTPCall(t *testing.T) {
 		})
 	}
 }
+
+type TestBody struct {
+	msg         string
+	ctxOverride context.Context
+	hook        string
+	method      string
+	errMsg      string
+	ttlMin      time.Duration
+	ttlMax      time.Duration
+	wantCode    int
+	wantBody    []byte // If nil, uses the request body
+
+	useHTTP2 bool
```

Comment:
let's not pass this flag here, instead let's have 2 test function - 

- TestHTTPCall(...)
- TestHTTP2Call(...)

both should get same set of test cases but http2 one will just pass flag when making call